### PR TITLE
React : Fixed overview

### LINKED ISSUE #29039: React (multiple): Some Lesson Overview items breaking TOP009 linting rule
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### Describe your suggestion

Our [TOP009 custom linting rule](https://github.com/TheOdinProject/curriculum/blob/main/markdownlint/docs/TOP009.md) requires Lesson Overview items to be statements and not questions. They must also end with a period.

Some of our lessons in the React course have Lesson Overview items that are either missing a period at the end, or are written as questions. These will need to have punctuation fixed, and any questions rephrased as statements.

## Acceptance criteria

- [ ] The following lessons have their Lesson Overview items fixed according to our TOP009 linting rule:
  - [Introduction To React](https://www.theodinproject.com/lessons/node-path-react-new-introduction-to-react)
  - [React Components](https://www.theodinproject.com/lessons/node-path-react-new-react-components)
  - [Keys In React](https://www.theodinproject.com/lessons/node-path-react-new-keys-in-react)
  - [How To Deal With Side Effects](https://www.theodinproject.com/lessons/node-path-react-new-how-to-deal-with-side-effects)
  - [Component Lifecycle Methods](https://www.theodinproject.com/lessons/node-path-react-new-component-lifecycle-methods)
  - [React Router](https://www.theodinproject.com/lessons/node-path-react-new-react-router)
  - [Styling React Applications](https://www.theodinproject.com/lessons/node-path-react-new-styling-react-applications)

Please comment below if you'd like to be assigned to this issue. Please do not open a PR until you have been assigned by a maintainer.

### Path

Ruby / Rails, Node / JS

### Lesson Url

As above

### (Optional) Discord Name

_No response_

### (Optional) Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by devnchill**
## Because

Some overview points didn't abide the rules , this fixed them .

## Issue

Closes #29039 

## Pull Request Requirements

-   [X ] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [X ] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [X ] The `Because` section summarizes the reason for this PR
-   [ X] The `This PR` section has a bullet point list describing the changes in this PR
-   [ X] If this PR addresses an open issue, it is linked in the `Issue` section
-   [ ] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [ X] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: react/getting_started_with_react/keys_in_react.md [hunk 1]
```diff
@@ -6,8 +6,8 @@ In this lesson, we will cover keys in React. Keys are special props for our comp
 
 This section contains a general overview of topics that you will learn in this lesson.
 
-- What are keys and why does React need them?
-- What are good and bad examples of keys?
+- Learn what are keys and their need in React.
```
Comment (MaoShizhong) [inline react/getting_started_with_react/keys_in_react.md:9]:
```suggestion
- Learn what keys are in React and why it needs them.
```

File: react/getting_started_with_react/react_components.md [hunk 2]
```diff
@@ -6,9 +6,9 @@ In this lesson we'll be going over the basics of React components - what they do
 
 This section contains a general overview of topics that you will learn in this lesson.
 
-- What are components?
-- How are components created?
-- Where do components live?
+- Learn about Reach components.
+- Understand how to create Components. 
+- Find where do components reside in a React project.
```
Comment (MaoShizhong) [inline react/getting_started_with_react/react_components.md:11]:
```suggestion
- Learn about React components.
- Understand how to create components.
- Describe where components reside in a React project.
```

File: react/introduction/introduction_to_react.md [hunk 3]
```diff
@@ -8,7 +8,7 @@ This section contains a general overview of topics that you will learn in this l
 
 - Understand what React is
```
Comment (MaoShizhong) [inline react/introduction/introduction_to_react.md:9]:
```suggestion
- Understand what React is.
```

File: react/introduction/introduction_to_react.md [hunk 4]
```diff
@@ -8,7 +8,7 @@ This section contains a general overview of topics that you will learn in this l
...
 
-### What is React?
+### Get familiar with React and its key features.
```
Comment (MaoShizhong) [inline react/introduction/introduction_to_react.md:11]: This heading doesn't need changing, only the lesson overview item as above.

```suggestion
### What is React?
```

File: react/the_react_ecosystem/react_router.md [hunk 5]
```diff
@@ -6,13 +6,12 @@ Up until this point in the curriculum, we have been building one-page applicatio
...
-- How do you add a "catch-all" route?
-- How do you pass data from a parent to any child components rendered via an outlet?
-- How do you add protected routes?
-
+- Understand what client-side routing is.
+- Learn how to use React Router for client-side routing.
+- Explore how to create nested and dynamic paths in React Router.
+- Discover how to add a 'catch-all' route in React Router.
+- Learn how to pass data from a parent component to any child components rendered via an outlet.
+- Understand how to implement protected routes in React Router.
```
Comment (MaoShizhong) [inline react/the_react_ecosystem/react_router.md:14]:
```suggestion
- Understand how to implement protected routes in React Router.

```

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
Just some linting and small word fixes for grammar/flow, but otherwise thank you for sorting these out.

After you've made the changes below, could you also [run the `fix:lesson` script](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md#curriculum-linting) on the "Introduction to React" file please? That should fix the remaining lint errors there.

File: react/the_react_ecosystem/react_router.md (hunk 5 above)
Comment (devnchill) [reply react/the_react_ecosystem/react_router.md:14]: What's the correction here ? They both look the sasme :sweat:

Comment (MaoShizhong) [reply react/the_react_ecosystem/react_router.md:14]: There's a blank line after the list you accidentally removed.

File: react/getting_started_with_react/react_components.md [hunk 6]
```diff
@@ -6,9 +6,9 @@ In this lesson we'll be going over the basics of React components - what they do
 
 This section contains a general overview of topics that you will learn in this lesson.
 
-- What are components?
-- How are components created?
-- Where do components live?
+- Learn about React components.
+- Understand how to create components.
+- Describe where components reside in a React projects.
```
Comment (MaoShizhong) [inline react/getting_started_with_react/react_components.md:11]:
```suggestion
- Describe where components reside in a React project.
```

**Review by MaoShizhong — CHANGES_REQUESTED** [review]

File: react/the_react_ecosystem/react_router.md (hunk 5 above)
Comment (devnchill) [reply react/the_react_ecosystem/react_router.md:14]: Oh got it

**Review by MaoShizhong — APPROVED** [review]

**Pull request merged**